### ISSUE TITLE
Unknown layer type: ImageGtData

### ISSUE THREAD

**Issue by PingDong2017** [zhaoweicai/mscnn#44]
I am trying to retrain the model just following the guide, using the KITTI data and the label. The following error occurred and stop the training process.

I0124 22:41:18.449167 22523 layer_factory.hpp:76] Creating layer data
F0124 22:41:18.449210 22523 layer_factory.hpp:80] Check failed: registry.count(type) == 1 (0 vs. 1) Unknown layer type: ImageGtData (known types: AbsVal, Accuracy, ArgMax, BNLL, BatchNorm, BatchReindex, BboxAccuracy, Bias, BoxOutput, Concat, ContrastiveLoss, Convolution, Data, Deconvolution, Detection, DetectionAccuracy, Dropout, DummyData, Eltwise, Embed, EuclideanLoss, Exp, Filter, Flatten, HDF5Data, HDF5Output, HingeLoss, Im2col, InfogainLoss, InnerProduct, LRN, Log, MVN, MemoryData, MultinomialLogisticLoss, PReLU, Pooling, Power, ProposalTarget, ROIPooling, ReLU, Reduction, Reshape, SPP, Scale, Sigmoid, SigmoidCrossEntropyLoss, Silence, Slice, SmoothL1Loss, Softmax, SoftmaxWithLoss, SoftmaxWithWeightedLoss, Split, TanH, Threshold, Tile)
*** Check failure stack trace: ***
    @     0x7effabe31daa  (unknown)
    @     0x7effabe31ce4  (unknown)
    @     0x7effabe316e6  (unknown)
    @     0x7effabe34687  (unknown)
    @     0x7effac494e03  caffe::LayerRegistry<>::CreateLayer()
    @     0x7effac4997ed  caffe::Net<>::Init()
    @     0x7effac49b218  caffe::Net<>::Net()
    @     0x7effac5fe852  caffe::Solver<>::InitTrainNet()
    @     0x7effac5ffbe8  caffe::Solver<>::Init()
    @     0x7effac5fff39  caffe::Solver<>::Solver()
    @     0x7effac47ee53  caffe::Creator_SGDSolver<>()
    @           0x408c82  train()
    @           0x405a88  main
    @     0x7effab130f45  (unknown)
    @           0x406107  (unknown)
    @              (nil)  (unknown)
Aborted (core dumped)

Could anyone check what is wrong? I compiled (with cuDNN 3 as advised) and installed the code without any problem before doing the training.

**Comment by PingDong2017** [comment]
The latest version (updated around Jan 20, 2017) does not have this problem.

**Comment by PingDong2017** [comment]
With OPENCV option on, this error happened again. Frustrated and will give it up.

**Comment by skyforwork** [comment]
@zhaoweicai  I have the same problem.How to solve it ?

**Comment by zhaoweicai** [comment]
I don't know why this error happens. Maybe "caffe.proto" is not updated? If you use the latest version, I think this error shouldn't happen, and it has nothing to do with OpenCV.

**Comment by buaa1406LiuZheng** [comment]
I met the same problem at first, and it does seem that it's relevant to the OpenCV.
If you set USE_OPENCV := 0 in the Makefile.config and it will not define USE_OPENCV in compiling. And if you see the source code image_gt_data_layer.cpp, you will find that if USE_OPENCV is undefined, this source file contains nothing, and therefore image_gt_data_layer has not been compiled.
I solved this problem by turning on the OpenCV.
